b2share/modules/records/indexer.py

```diff
@@ -48,6 +48,10 @@ def record_to_index(record):
 def indexer_receiver(sender, json=None, record=None, index=None,
                      **dummy_kwargs):
     """Connect to before_record_index signal to transform record for ES."""
+    if is_deposit(record.model):
+        if 'external_pids' in json['_deposit']:
+            json['external_pids'] = json['_deposit']['external_pids']
```

Comment:
We should be careful to not expose the files "key" if the record is under embargo. This might contain private information. This is why we already remove the `_files` field.